bot/exts/halloween/spooknamerate.py

```diff
@@ -0,0 +1,371 @@
+                    party = [random.choice([':partying_face:', ':tada:']) for _ in range(random.randint(1, 10))]
+                    await channel.send(" ".join(party))
+
+                else:
+                    # no participants in current iteration
+                    await channel.send('Hmm... Looks like no one participated! :cry:')
+
+            async with self.checking_messages:  # Acquire the lock to delete the messages
+                self.messages = {}  # reset the messages
+
+        # send the next name
+        self.name = f"{random.choice(self.first_names)} {random.choice(self.last_names)}"
+        self.data['name'] = self.name
+        self.update_json()
+
+        await channel.send("Let's move on to the next name!\nAnd the next name is...\n"
+                           f"**{self.name}**!\nTry to spookify that... :smirk:")
+
+        self.poll = False  # accepting responses
+
+    @announce_name.before_loop
+    async def wait_till_scheduled_time(self) -> None:
+        """Waits till the next day's 12AM if crossed it, otherwise waits till the same day's 12AM."""
+        now = datetime.utcnow()
+        if now.hour < 12:
+            twelve_pm = now.replace(hour=12, minute=0, seconds=0, microsecond=0)
+            time_left = twelve_pm - now
+            await asyncio.sleep(time_left.seconds)
+            return
+
+        tomorrow_12pm = now + timedelta(days=1)
+        tomorrow_12pm = tomorrow_12pm.replace(hour=12, minutes=0, second=0, microsecond=0)
+        await asyncio.sleep((tomorrow_12pm - now).seconds)
+
+    async def get_responses_list(self, final: bool = False) -> Embed:
+        """Returns an embed containing the responses of the people."""
+        channel = await self.get_channel()
+
+        embed = Embed(
+            color=Colours.red,
+            title="",
+            description=""
+        )
+
+        if self.messages:
+            if final:
+                embed.title = "Spook Name Rate is about to end!"
+                embed.description = ("This Spook Name Rate round is about to end in 2 hours! You can review "
+                                     "the entries below! Have you rated other's names?")
+            else:
+                embed.title = "All the spookified names!"
+                embed.description = "See a list of all the entries entered by everyone!"
+        else:
+            embed.title = "No one has added an entry yet..."
+
+        for message_id, data in self.messages.items():
+
+            embed.add_field(
+                name=(await self.bot.fetch_user(data['author'])).name,
+                value=f"[{data['name']}](https://discord.com/channels/{Client.guild}/{channel.id}/{message_id})",
+            )
+
+        return embed
+
+    async def get_channel(self) -> TextChannel:
+        """Gets the seasonalbot-channel after waiting until ready."""
+        await self.bot.wait_until_ready()
+        return self.bot.get_channel(self.channel_id)
```

Comment:
This method can also return None if the channel is not found, so you could update the function annotation and a check for that.